src/workerd/api/sql.c++

```diff
@@ -19,6 +19,11 @@ jsg::Ref<SqlStorage::Cursor> SqlStorage::exec(jsg::Lock& js, kj::String querySql
   return jsg::alloc<Cursor>(*sqlite, regulator, querySql, kj::mv(bindings));
 }
 
+kj::StringPtr SqlStorage::ingest(jsg::Lock& js, kj::String querySql) {
```

Comment:
Returning `StringPtr` here is not safe since it's going to be pointing into `querySql` which is deleted at the end of this function, before the caller can do anything with the return value.

Reply:
For my own education, is there any reason why we need to take querySql as a `kj::String`?  If I were doing this just in KJ-land, it would make sense to take in `kj::StringPtr` and return a `kj::StringPtr` (with the documented restriction that the returned `kj::StringPtr` has the same lifetime as the input `kj::StringPtr`), but I don't know if the runtime has other restrictions.

Reply:
This is a limitation of the JSG layer. We don't support receiving a StringPtr from JavaScript because there's nothing for it to point to: the original JS string is in UTF-16. Since we have to make a copy anyway, we make you accept ownership of that copy.

I _suppose_ we could change that and _not_ make you take ownership, but it'd be precarious in this particular situation, where you're trying to return a substring of that string... you have to count on the JSG layer converting that _back_ to JS before the input string is freed.